src/sentry/integrations/pagerduty/integration.py

```diff
@@ -0,0 +1,166 @@
+from __future__ import absolute_import
+
+from django.utils.translation import ugettext_lazy as _
+from django.db import transaction
+
+from sentry import options
+
+from sentry.utils import json
+from sentry.integrations import (
+    IntegrationInstallation,
+    IntegrationFeatures,
+    IntegrationMetadata,
+    IntegrationProvider,
+    FeatureDescription,
+)
+from sentry.models import PagerDutyServiceProject, Project
+from sentry.pipeline import PipelineView
+
+DESCRIPTION = """
+PagerDuty Description
+"""
+
+FEATURES = [
+    FeatureDescription(
+        """
+        Configure rule based PagerDuty notifications!!
+        """,
+        IntegrationFeatures.ALERT_RULE,
+    )
+]
+
+metadata = IntegrationMetadata(
+    description=_(DESCRIPTION.strip()),
+    features=FEATURES,
+    author="The Sentry Team",
+    noun=_("Installation"),
+    issue_url="https://github.com/getsentry/sentry/issues/new?title=PagerDuty%20Integration:%20&labels=Component%3A%20Integrations",
+    source_url="https://github.com/getsentry/sentry/tree/master/src/sentry/integrations/pagerduty",
+    aspects={},
+)
+
+
+class PagerDutyIntegration(IntegrationInstallation):
+    def get_client(self):
+        pass
+
+    def get_organization_config(self):
+        from sentry.models import Project
+
+        projects = Project.objects.filter(
+            organization_id__in=self.model.organizations.values_list("id", flat=True)
+        )
+        items = []
+        for p in projects:
+            items.append({"value": p.id, "label": p.name})
+
+        service_options = [(s["id"], s["name"]) for s in self.services.values()]
+
+        fields = [
+            {
+                "name": "project_mapping",
+                "type": "choice_mapper",
+                "label": "Map projects in Sentry to services in PagerDuty",
+                "help": "When an alert rule is triggered in a project, this mapping will let us know what service to create the incident in PagerDuty.",
+                "addButtonText": "Add Sentry Project",
+                "addDropdown": {
+                    "emptyMessage": "All projects configured",
+                    "noResultsMessage": "Could not find project",
+                    "items": items,
+                },
+                "mappedSelectors": {
+                    "service": {"choices": service_options, "placeholder": "Select a service"}
+                },
+                "columnLabels": {"service": "Service"},
+                "mappedColumnLabel": "Sentry Project",
+            }
+        ]
+
+        return fields
+
+    def update_organization_config(self, data):
+
+        if "project_mapping" in data:
+            project_ids_and_services = data.pop("project_mapping")
```

Comment:
@MeredithAnya Is the intention of popping here to mutate the input dictionary `data`?

Reply:
tbh this was copy pasta from the vsts integration, but I'm assuming so yeah.